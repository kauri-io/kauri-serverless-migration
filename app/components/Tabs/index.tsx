import * as React from "react";
import styled from "styled-components";

interface ITheme {
  bg: { [val: string]: string };
  space: {
    [val: number]: number;
  };
}

interface ITabContainerProps {
  minWidth?: string;
}

const TabContainer = styled.div`
  display: flex;
  flex-direction: column;
  overflow-y: auto;
  min-width: ${(props: ITabContainerProps) => props.minWidth};
  padding-bottom: ${props => props.theme.space[2]}px;
`;

interface ITabsProps {
  dark?: boolean;
  padContent: boolean;
  centerTabs?: boolean;
  bg: string;
  router?: any;
}
const Tabs = styled.div<ITabsProps>`
  height: 50px;
  width: 100%;
  color: ${props =>
    props.dark ? "white" : props.theme.colors.primaryTextColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${props =>
    props.dark ? props.theme && props.theme.bg[props.bg] : "transparent"};
  ${props =>
    props.padContent &&
    `padding: 0px calc((100vw - ${props.theme.breakpoints[2]}) / 2)`};
  ${props => props.centerTabs && "justify-content: center"};
  > :not(:last-child) {
    margin-right: ${props => props.theme && props.theme.space[1]}px;
  }
  @media (max-width: ${props => props.theme.breakpoints[0]}) {
    padding: 0px ${props => props.theme.space[1]}px;
  }
`;

interface ITabProps {
  selected: boolean;
  theme?: ITheme;
  minWidth?: string;
}

const Tab = styled.div<ITabProps>`
  height: 50px;
  display: flex;
  align-items: center;
  border-bottom: ${props => (props.selected ? "4px solid #0ba986" : "none")};
  padding-top: ${props => (props.selected ? "4px" : 0)};
  cursor: pointer;
  font-weight: 700;
  text-transform: uppercase;
  font-size: ${props => props.theme.fontSizes[1]}px;
`;

interface ITab {
  name: string;
  callback?: (args?: any) => void;
}

interface IProps {
  tabs: Array<ITab | null>;
  panels: Element[] | JSX.Element[];
  padContent?: boolean;
  centerTabs?: boolean;
  bg?: string;
  minWidth?: string;
  dark?: boolean;
  router?: any;
  passChangeTabFunction?: (func: any) => void;
}

interface IState {
  selectedTabIndex: number;
}

class TabsComponent extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      selectedTabIndex: 0,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  public componentDidMount() {
    if (this.props.passChangeTabFunction) {
      this.props.passChangeTabFunction(this.changeTab.bind(this));
    }
    // Context - homepage tab is null for communities if empty and not admin soo
    if (
      Array.isArray(this.props.tabs) &&
      this.props.tabs.length &&
      !this.props.tabs[0]
    ) {
      this.changeTab(1);
    }
  }

  public changeTab(index: number) {
    this.setState({ selectedTabIndex: index });
  }

  public handleClick(index: number, tab: ITab) {
    this.changeTab(index);
    if (tab.callback) {
      tab.callback();
    }
  }

  public render() {
    const props = this.props;
    const {
      padContent = true,
      bg = "bgSecondary",
      minWidth,
      centerTabs,
      dark,
    } = this.props;

    return (
      <TabContainer className="tabs" minWidth={minWidth}>
        <Tabs
          dark={dark}
          bg={bg}
          padContent={padContent}
          centerTabs={centerTabs}
        >
          {this.props.tabs.map(
            (tab, index) =>
              tab && (
                <Tab
                  key={index}
                  minWidth={minWidth}
                  onClick={() => this.handleClick(index, tab)}
                  selected={index === this.state.selectedTabIndex}
                >
                  {tab.name}
                </Tab>
              )
          )}
        </Tabs>
        {props.panels[this.state.selectedTabIndex]}
      </TabContainer>
    );
  }
}

export default TabsComponent;
