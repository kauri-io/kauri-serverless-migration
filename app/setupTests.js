import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
