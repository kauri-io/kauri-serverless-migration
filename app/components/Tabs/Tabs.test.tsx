import Tabs from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    tabs: [
        { name: 'First', callback: jest.fn() },
        { name: 'second', callback: jest.fn() },
    ],
    panels: [<div>Test</div>, <div>Test 2</div>],
}

describe('components/Tabs', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<Tabs {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
