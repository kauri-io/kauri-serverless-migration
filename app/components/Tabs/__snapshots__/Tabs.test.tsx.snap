// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Tabs should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <TabsComponent
    panels={
      Array [
        <div>
          Test
        </div>,
        <div>
          Test 2
        </div>,
      ]
    }
    tabs={
      Array [
        Object {
          "callback": [MockFunction],
          "name": "First",
        },
        Object {
          "callback": [MockFunction],
          "name": "second",
        },
      ]
    }
  >
    <Tabs__TabContainer
      className="tabs"
    >
      <StyledComponent
        className="tabs"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Tabs__TabContainer-sc-1cx8ezj-0",
              "isStatic": false,
              "lastClassName": "bzNgfs",
              "rules": Array [
                "display:flex;flex-direction:column;overflow-y:auto;min-width:",
                [Function],
                ";padding-bottom:",
                [Function],
                "px;",
              ],
            },
            "displayName": "Tabs__TabContainer",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Tabs__TabContainer-sc-1cx8ezj-0",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="tabs Tabs__TabContainer-sc-1cx8ezj-0 bzNgfs"
        >
          <Tabs
            bg="bgSecondary"
            padContent={true}
          >
            <StyledComponent
              bg="bgSecondary"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Tabs-sc-1cx8ezj-1",
                    "isStatic": false,
                    "lastClassName": "esitTx",
                    "rules": Array [
                      "height:50px;width:100%;color:",
                      [Function],
                      ";display:flex;flex-direction:row;align-items:center;background-color:",
                      [Function],
                      ";",
                      [Function],
                      ";",
                      [Function],
                      ";>:not(:last-child){margin-right:",
                      [Function],
                      "px;}@media (max-width:",
                      [Function],
                      "){padding:0px ",
                      [Function],
                      "px;}",
                    ],
                  },
                  "displayName": "Tabs",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Tabs-sc-1cx8ezj-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              padContent={true}
            >
              <div
                className="Tabs-sc-1cx8ezj-1 esitTx"
              >
                <Tabs__Tab
                  key="0"
                  onClick={[Function]}
                  selected={true}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Tabs__Tab-sc-1cx8ezj-2",
                          "isStatic": false,
                          "lastClassName": "gaDbSs",
                          "rules": Array [
                            "height:50px;display:flex;align-items:center;border-bottom:",
                            [Function],
                            ";padding-top:",
                            [Function],
                            ";cursor:pointer;font-weight:700;text-transform:uppercase;font-size:",
                            [Function],
                            "px;",
                          ],
                        },
                        "displayName": "Tabs__Tab",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Tabs__Tab-sc-1cx8ezj-2",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={true}
                  >
                    <div
                      className="Tabs__Tab-sc-1cx8ezj-2 bliwJC"
                      onClick={[Function]}
                      selected={true}
                    >
                      First
                    </div>
                  </StyledComponent>
                </Tabs__Tab>
                <Tabs__Tab
                  key="1"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Tabs__Tab-sc-1cx8ezj-2",
                          "isStatic": false,
                          "lastClassName": "gaDbSs",
                          "rules": Array [
                            "height:50px;display:flex;align-items:center;border-bottom:",
                            [Function],
                            ";padding-top:",
                            [Function],
                            ";cursor:pointer;font-weight:700;text-transform:uppercase;font-size:",
                            [Function],
                            "px;",
                          ],
                        },
                        "displayName": "Tabs__Tab",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Tabs__Tab-sc-1cx8ezj-2",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="Tabs__Tab-sc-1cx8ezj-2 gaDbSs"
                      onClick={[Function]}
                      selected={false}
                    >
                      second
                    </div>
                  </StyledComponent>
                </Tabs__Tab>
              </div>
            </StyledComponent>
          </Tabs>
          <div>
            Test
          </div>
        </div>
      </StyledComponent>
    </Tabs__TabContainer>
  </TabsComponent>
</ThemeProvider>
`;
